BlockchainPatterns/token.rho

```diff
@@ -0,0 +1,173 @@
+// LICENSE: Anyone can freely read and learn from this code
+// but it may not be deployed in a public blockchain until I
+// am paid a reasonable amount for my time writing it.
+// Then it will be Apache-2.0
+
+// Based rougly on http://erights.org/elib/capability/ode/ode-capabilities.html#simple-money
+// and https://github.com/rchain/rchain/blob/dev/casper/src/main/rholang/MakeMint.rho
+
+new tokenFactory in {
+  contract tokenFactory (@supply, log, resolve) = {
+    log!("Inside token factory. Haven't yet validated supply") |
+    if (supply < 0) {
+      log!("Initial balance must be positive")
+    }
+    else {
+      new mintId, coinFactory in {
+
+        // Return a single coin with total supply to creator
+        coinFactory!(supply, *resolve) |
+
+        // This implementation does not leak the raw MintId.
+        // If you want centralized monetary policy, include
+        // something like the following
+        // resolve!(*mintId) // CAUTION: Allows unilateral inflation
+
+        contract coinFactory (@value, return) = {
+          log!("coinFactory called") |
+
+          new coinId, unspentCh in {
+            // Initialize coin, and give it to its owner
+            unspentCh!(Nil) |
+            //ISSUE! This is subject to destructuring!
+            //return!(bundle!{*mintId | *coinId}) |
+            return!(*mintId | *coinId) |
+
+            // Methods use compound names. If you prefer a first-arg dispatching
```

Comment:
when sending on public names is disabled, will methods using compound names still work?

Reply:
I'm unsure whether that's still on any roadmap, but even if it is my understanding is that only forgeable names (names that do not depend on any particular unforgeable name being in scope) would be disabled.

I don't have any reference to point at, but I think it is the most reasonable notion of "public" names.